test/range/container/constexpr_string_test.cpp

```diff
@@ -0,0 +1,249 @@
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ==========================================================================
+
+#include <gtest/gtest.h>
+
+#include <seqan3/range/container/constexpr_string.hpp>
+#include <seqan3/range/container/concept.hpp>
+#include <seqan3/range/concept.hpp>
+
+#include <utility>
+#include <string>
+
+using namespace std::literals;
+using namespace seqan3;
+
+// standard construction.
+TEST(constexpr_string, standard_construction)
+{
+    EXPECT_TRUE((std::is_default_constructible_v<constexpr_string<4>>));
+    EXPECT_TRUE((std::is_trivially_default_constructible_v<constexpr_string<4>>));
+    EXPECT_TRUE((std::is_nothrow_default_constructible_v<constexpr_string<4>>));
+    EXPECT_TRUE((std::is_copy_constructible_v<constexpr_string<4>>));
+    EXPECT_TRUE((std::is_trivially_copy_constructible_v<constexpr_string<4>>));
+    EXPECT_TRUE((std::is_nothrow_copy_constructible_v<constexpr_string<4>>));
+    EXPECT_TRUE((std::is_move_constructible_v<constexpr_string<4>>));
+    EXPECT_TRUE((std::is_trivially_move_constructible_v<constexpr_string<4>>));
+    EXPECT_TRUE((std::is_nothrow_move_constructible_v<constexpr_string<4>>));
+    EXPECT_TRUE((std::is_copy_assignable_v<constexpr_string<4>>));
+    EXPECT_TRUE((std::is_trivially_copy_assignable_v<constexpr_string<4>>));
+    EXPECT_TRUE((std::is_nothrow_copy_assignable_v<constexpr_string<4>>));
+    EXPECT_TRUE((std::is_move_assignable_v<constexpr_string<4>>));
+    EXPECT_TRUE((std::is_trivially_move_assignable_v<constexpr_string<4>>));
+    EXPECT_TRUE((std::is_nothrow_move_assignable_v<constexpr_string<4>>));
+}
+
+TEST(constexpr_string, container_concept)
+{
+    EXPECT_TRUE(container_concept<constexpr_string<4>>);
+    EXPECT_TRUE(random_access_range_concept<constexpr_string<4>>);
+}
+
+// construction from literal.
+TEST(constexpr_string, construct_from_literal)
+{
+     EXPECT_TRUE((std::is_same_v<constexpr_string<5>, decltype(constexpr_string{"hello"})>));
+}
+
+// construction from char.
+TEST(constexpr_string, construct_from_char)
+{
+     EXPECT_TRUE((std::is_same_v<constexpr_string<1>, decltype(constexpr_string{'h'})>));
+}
+
+// construction from array.
+TEST(constexpr_string, construct_from_array)
+{
+     EXPECT_TRUE((std::is_same_v<constexpr_string<5>,
+                                 decltype(constexpr_string{std::array{'h','e','l','l','o'}})>));
+}
+
+template <size_t S>
+struct foo
+{
+    inline size_t get() const {return S;};
+};
```

Comment:
what's the point of this? If you want to check if a variable is constexpr, why not just create a `constexpr auto = foo.size()` to check?

Reply:
??? This would be essentially the same, but just with a different interface name, or am I missing something here.
Then why would that be relevant for the test?

Reply:
Ok it must be constexpr context if it is a constexpr variable. Oh fu.. great stuff!